### DISCUSSION TITLE
How to debug serverless function 500 internal server error

### DISCUSSION THREAD

**Discussion by xmflsct** [vercel/vercel#3978] [General] [answered]
One of my builds did accept api calls, and that 500 error shows that it is not internal server error but an error of the script.

But after that, no matter what I deploy, I always only get simple 500 internal server error, and the dashboard shows no requests made.
How can I debug this case? I try other path like `/api/non-exist` and it gives 404 which is correct.

**Comment by styfle** [comment, marked as answer]
Hi @xmflsct 

Runtime logs aren't stored so you'll have to have the dashboard open to your function's runtime logs, then visit the function in another tab.

If it throws an error, that will persist in the error log.

If you need permanent runtime logs you can set up Log Drains with a 3rd party logging provider.

https://zeit.co/blog/refined-logging

**Comment by styfle** [comment]
You can also run functions locally with `now dev`.

https://zeit.co/docs/v2/serverless-functions/introduction#local-development

**Reply by xmflsct** [reply]
Thank you! This question is related to my other question #3977, which I have figured out how to do it, but now really why. :)

**Comment by Khushbu133** [comment]
I m getting this error Serverless Function 
500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED
ID: bom1::7jzq6-1665384130714-baa552278a8d
please help me

**Reply by amyegan** [reply]
Hi @Khushbu133! Have you been able to get any additional details from the [logs](https://vercel.com/docs/concepts/deployments/logs)? You might need to alter your Serverless Function to print out more error details to help you figure out what is going wrong.

**Comment by devgalib** [comment]
I m getting this error Serverless Function

500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED
ID: bom1::tlpcr-1665692001843-8318aaea793b
